lib/client.js

```diff
@@ -170,23 +211,48 @@ Client.prototype._invoke = function(method, args, location, callback, options, e
   self.lastMessage = message;
   self.lastRequest = xml;
 
-  req = http.request(location, xml, function(err, response, body) {
+  self.emit('message', message);
+  self.emit('request', xml);
+
+  req = self.httpClient.request(location, xml, function(err, response, body) {
     var result;
     var obj;
     self.lastResponse = body;
     self.lastResponseHeaders = response && response.headers;
+    self.emit('response', body);
 
     if (err) {
       callback(err);
-    } else if (response.statusCode !== 200) {
-      callback(new Error('Invalid response: ' + response.statusCode + '\nBody: ' + body));
     } else {
       try {
         obj = self.wsdl.xmlToObject(body);
+
+        if (!obj.Body && response.statusCode !== 200) {
```

Comment:
@jxayala and @y-tang - I just realized this is moot, because an equivalent snuck in from upstream to handle the case where we receive an unparsable body (e.g. HTML).  See: `Cannot parse response` below